src/GenSerialization.jl

```diff
@@ -1,5 +1,32 @@
 module GenSerialization
```

Comment:
The latest commit has all of the requisites: (a) README has demo, installation, and warnings. (b) Serialization has been stripped down to the naive one mentioned long ago. An experimental speed ups have been removed.

Once this PR is merged, we should update Gen.jl to have this as a package extension.